tests/TestCase/TestSuite/IntegrationTestTraitTest.php

```diff
@@ -158,11 +158,10 @@ public function testDataCastToString(): void
      */
     public function testRequestBuilding(): void
     {
+        $this->setJsonHeader();
```

Comment:
The name doesn't seem quite right to me. It sounds like you're setting the json header not marking the request as json type.

Reply:
We can for sure call it differently if we find a better name but as far as I know a JSON request is defined by those 2 headers. Thats why I am not really sure what you mean.

Reply:
What about something more descriptive of what the controller request will be doing? Currently we have methods like `enableCsrfToken()` and `enableRetainFlashMessages()`. What about `requestJson()` or `requestAsJson()`?